app/controllers/proposal_controller.rb

```diff
@@ -46,31 +46,28 @@ def create
                            event_role: 'submitter')
     @event.event_users.new(user: current_user,
                            event_role: 'speaker')
+    if @event.save
+      ahoy.track 'Event submission', title: 'New submission'
 
-    unless @event.save
+      flash[:notice] = 'Proposal was successfully submitted.'
+      redirect_to conference_program_proposal_index_path(@conference.short_title)
+    else
       flash[:error] = "Could not submit proposal: #{@event.errors.full_messages.join(', ')}"
       render action: 'new'
-      return
     end
-
-    ahoy.track 'Event submission', title: 'New submission'
-
-    flash[:notice] = 'Proposal was successfully submitted.'
-    redirect_to conference_program_proposal_index_path(@conference.short_title)
   end
 
   def update
     authorize! :update, @event
     @url = conference_program_proposal_path(@conference.short_title, params[:id])
 
-    if !@event.update(event_params)
+    if @event.update(event_params)
+      redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
+                  notice: 'Proposal was successfully updated.')
+    else
       flash[:error] = "Could not update proposal: #{@event.errors.full_messages.join(', ')}"
-      render action: 'new'
-      return
+      render action: 'edit'
     end
-
-    redirect_to(conference_program_proposal_index_path(conference_id: @conference.short_title),
-                notice: 'Proposal was successfully updated.')
   end
 
   def destroy
```

Comment:
Shouldn't this be named `withdraw`? This method also making `DELETE` request for updating the record when it should make `PATCH` request, not very RESTful.